test/e2e/kappcontroller/installation_namespace_test.go

```diff
@@ -0,0 +1,288 @@
+// Copyright 2023 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package kappcontroller
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/vmware-tanzu/carvel-kapp-controller/test/e2e"
+)
+
+func Test_AppInstallationNamespace(t *testing.T) {
+	env := e2e.BuildEnv(t)
+	logger := e2e.Logger{}
+	kapp := e2e.Kapp{t, env.Namespace, logger}
+	kubectl := e2e.Kubectl{t, env.Namespace, logger}
+	name := "app-installation-namespace"
+	installationNamespace := "installation-namespace"
+	installationNamespaceApp := "installation-namespace-app"
+
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+		kapp.Run([]string{"delete", "-a", installationNamespaceApp})
+	}
+	cleanUp()
+	defer cleanUp()
+
+	namespaceYAML := fmt.Sprintf(`---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: %s`, installationNamespace)
+
+	sas := e2e.ServiceAccounts{env.Namespace}
+	appYAML := fmt.Sprintf(`---
+apiVersion: kappctrl.k14s.io/v1alpha1
+kind: App
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  serviceAccountName: kappctrl-e2e-ns-sa
+  installationNamespace: %s
+  fetch:
+  - inline:
+      paths:
+        file.yml: |
+          apiVersion: v1
+          kind: ConfigMap
+          metadata:
+            name: my-cm
+          data:
+            key: value
+  template:
+  - ytt: {}
+  deploy:
+  - kapp: {}`, name, env.Namespace, installationNamespace) + sas.ForClusterYAML()
+
+	kapp.RunWithOpts([]string{"deploy", "-a", installationNamespaceApp, "-f", "-"}, e2e.RunOpts{StdinReader: strings.NewReader(namespaceYAML)})
+
+	_, err := kapp.RunWithOpts([]string{"deploy", "-a", name, "-f", "-"}, e2e.RunOpts{StdinReader: strings.NewReader(appYAML)})
```

Comment:
For new tests we have been trying to use `"github.com/stretchr/testify/assert"`

So this would be `assert.NoError(t, err)`

Reply:
Updated.